spec/lib/project_copier_spec.rb

```diff
@@ -0,0 +1,46 @@
+require 'spec_helper'
+
+describe ProjectCopier do
+  describe '#copy' do
+    let(:project) { create(:full_project, build_extra_contents: true)}
+    let(:copyist) { create(:user) }
+    let(:tags) { create(:tag, resource: project) }
```

Comment:
none of these associated records exist as they are never called before the copy. Might want to add a `let!(:tags)` or call them before you use them. Also you should probably test these associations exist on the copied project, i wouldn't validate the copy just that the included associations are copied.

Reply:
yup. bang!